File: data/code_pairs/fortran/2357.f90
MODULE DataPrecisionGlobals

  IMPLICIT NONE

  INTEGER, PARAMETER :: i32=SELECTED_INT_KIND(6)
  INTEGER, PARAMETER :: i64=SELECTED_INT_KIND(12)
  INTEGER, PARAMETER :: r32=KIND(1.0)
  INTEGER, PARAMETER :: r64=KIND(1.0D0)
  INTEGER, PARAMETER :: default_prec=r64
  REAL(r64), PARAMETER :: constant_zero=0.0d0
  REAL(r64), PARAMETER :: constant_one=1.0d0
  REAL(r64), PARAMETER :: constant_minusone=-1.0d0
  REAL(r64), PARAMETER :: constant_twenty=20.0d0
  REAL(r64), PARAMETER :: constant_pointfive=0.5d0
  REAL(r64), PARAMETER :: EXP_LowerLimit=-20.d0
  REAL(r64), PARAMETER :: EXP_UpperLimit= 40.d0

END MODULE DataPrecisionGlobals

File: data/code_pairs/cpp/2357.cpp
#ifndef DATA_PRECISION_GLOBALS_H
#define DATA_PRECISION_GLOBALS_H

#include <cstdint>

namespace DataPrecisionGlobals {

    using int32 = int32_t;
    using int64 = int64_t;

    using real32 = float;
    using real64 = double;
    using default_prec = real64;

    constexpr real64 constant_zero = 0.0;
    constexpr real64 constant_one = 1.0;
    constexpr real64 constant_minusone = -1.0;
    constexpr real64 constant_twenty = 20.0;
    constexpr real64 constant_pointfive = 0.5;
    constexpr real64 EXP_LowerLimit = -20.0;
    constexpr real64 EXP_UpperLimit = 40.0;

}

#endif // DATA_PRECISION_GLOBALS_H